ci-generate-build-n-test.js

```diff
@@ -0,0 +1,127 @@
+const fs = require("fs");
+
+const services = [
+  "auditlogs",
+  "create-artifact-worker",
+  "deployments",
+  "deviceauth",
+  "deviceconfig",
+  "deviceconnect",
+  "devicemonitor",
+  "inventory",
+  "mender-gateway",
+  "tenantadm",
+  "useradm",
+  "workflows",
+];
+
+const pipelineTemplate = (jobs) => `stages:
+  - build
+  - test
+${jobs.join("")}
+`;
+
+const getChangesRule = ({
+  RULES_CHANGES_COMPARE_TO_REF,
+  service,
+}) => `- changes:
+        paths: ["backend/services/${service}/**/*", "backend/pkg/**/*"]
+        compare_to: "${RULES_CHANGES_COMPARE_TO_REF}"`;
+
+const buildTemplate = ({
+  CI_COMMIT_REF_NAME,
+  CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX,
+  CI_PIPELINE_ID,
+  CI_REGISTRY_IMAGE,
+  DOCKER_PUBLISH_COMMIT_TAG,
+  RULES_CHANGES_COMPARE_TO_REF,
+  service,
+}) => `
+build:${service}:docker-multiplatform:
+  image: "registry.gitlab.com/northern.tech/mender/mender-test-containers:docker-multiplatform-buildx-v1-master"
+  stage: build
+  rules:
+    ${getChangesRule({ RULES_CHANGES_COMPARE_TO_REF, service })}
+    - if: '$CI_COMMIT_TAG =~ /^saas-[a-zA-Z0-9.]+$/'
+      when: never
+    - when: on_success
+  tags:
+    - hetzner-amd-beefy
+  services:
+    - name: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/docker:20.10.21-dind
```

Comment:
This version of docker is over 3 years old. Could we make the docker version more transparent in the global `variable` section?
```suggestion
    - name: ${CI_DEPENDENCY_PROXY_DIRECT_GROUP_IMAGE_PREFIX}/docker:${DOCKER_VERSION}-dind
```